### PR TITLE
0.24.2: stop vent from swallowing and use patched Tendermint

### PR DESCRIPTION
**Pull request by silasdavis**

### CODE REVIEW

File: execution/evm/abi/core.go [hunk 1]
```diff
@@ -169,25 +170,27 @@ func LoadPath(abiFileOrDir string) (*AbiSpec, error) {
...
-		if err != nil {
-			return fmt.Errorf("error returned while walking abiDir '%s': %v", abiFileOrDir, err)
-		}
-		ext := filepath.Ext(path)
-		if fi.IsDir() || !(ext == ".bin" || ext == ".abi") {
-			return nil
-		}
-		if err == nil {
-			abiSpc, err := ReadAbiSpecFile(path)
+	for _, dir := range strings.Split(abiFileOrDir, ":") {
```
Comment (seanyoung) [inline execution/evm/abi/core.go]: This should be ; on windows else "burrow vent start --abi c:\abis" is not going to work.

File: vent/sqlsol/projection.go [hunk 2]
```diff
@@ -40,39 +40,41 @@ func NewProjectionFromBytes(bs []byte) (*Projection, error) {
...
 
 	const errHeader = "NewProjectionFromFolder():"
 
-	err := filepath.Walk(fileOrDir, func(path string, _ os.FileInfo, err error) error {
-		if err != nil {
-			return fmt.Errorf("error walking event spec files location '%s': %v", fileOrDir, err)
-		}
-		if filepath.Ext(path) == ".json" {
-			bs, err := readFile(path)
+	for _, dir := range strings.Split(specFileOrDir, ":") {
```
Comment (seanyoung) [inline vent/sqlsol/projection.go]: same here

File: execution/evm/vm_test.go [hunk 3]
```diff
@@ -689,6 +689,7 @@ func TestRevert(t *testing.T) {
 
 	output, cErr := ourVm.Call(cache, NewNoopEventSink(), account1, account2, bytecode, []byte{}, 0, &gas)
 	assert.Error(t, cErr, "Expected execution reverted error")
+	fmt.Println(cErr)
```
Comment (gregdhill) [inline execution/evm/vm_test.go]: Leftover? ^

File: genesis/genesis.go [hunk 4]
```diff
@@ -68,13 +69,14 @@ type params struct {
...
-	hash              []byte
-	Params            params `json:",omitempty" toml:",omitempty"`
-	Salt              []byte `json:",omitempty" toml:",omitempty"`
+	AppHash           binary.HexBytes `json:",omitempty" toml:",omitempty"`
+	Params            params          `json:",omitempty" toml:",omitempty"`
+	Salt              []byte          `json:",omitempty" toml:",omitempty"`
 	GlobalPermissions permission.AccountPermissions
 	Accounts          []Account
 	Validators        []Validator
+	// memo
```
Comment (gregdhill) [inline genesis/genesis.go]: Leftover comment?

File: integration/rpcevents/execution_events_server_test.go [hunk 5]
```diff
@@ -184,6 +184,7 @@ func TestRevert(t *testing.T) {
 	contractAddress := txe.Receipt.ContractAddress
 	txe = rpctest.CallContract(t, tcli, inputAddress, contractAddress, data)
 	assert.Equal(t, errors.ErrorCodeExecutionReverted, txe.Exception.Code)
+	assert.Contains(t, txe.Exception.Error(), "I have reverted")
```
Comment (gregdhill) [inline integration/rpcevents/execution_events_server_test.go]: Is this the correct error message?

Comment (silasdavis) [reply integration/rpcevents/execution_events_server_test.go]: yep but good spot

**Pull request merged**